### DISCUSSION TITLE
ESP32-CAM

### DISCUSSION THREAD

**Discussion by und3f1ned** [hippo5329/linorobot2_hardware#39] [General]
Is ESP32-CAM supported? If yes, is it feasible to set up a stream transport for the camera (similar to how it's done with a LiDAR)? Or is this beyond its performance capabilities?

**Comment by hippo5329** [comment]
Esp32-cam is not supported. It has too few pinouts. Esp32 is cheap. You can have one esp32 for linorobot2_hardware and another esp32-cam for esp32cam-rtsp. And process the rtsp video stream on your PC host.

https://github.com/rzeldent/esp32cam-rtsp

**Reply by und3f1ned** [reply]
Thank you for the clarification! While the ESP32 boards are undoubtedly inexpensive, my intention was to manage everything with a single microcontroller for a very small robot without a LiDAR. Using two controllers in this scenario feels more like a workaround than an elegant solution. Appreciate the suggestion, though!

**Comment by hippo5329** [comment]
If you only need to move the robot without wheel odometry, then only 4 pwm pins , 12 13 14 16, on esp32-cam are needed to control the motor drivers. You can move the robot with teleop/twist. You may use 2 4 15 1 3 for I2C/IMU, LED or ultra-sonic.

**Comment by hippo5329** [comment]
I did not decode the Lidar data. It should be possible to send the camera video stream, too. The wifi on esp32 can transfer several Mbits per seconds normally.
